### PR TITLE
Create Index Builder

### PR DESCRIPTION
**Pull request by DeusAnimo**

### CODE REVIEW

**Comment by DeusAnimo** [comment]
Возможно нужен еще некий функционал, например удаление дескриптора из графа (поведение для основного контейнера `IndexBuilder` и индекса `Dynamic index` разное)

File: lunavl/sdk/indexes/builder.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+    def append(self, descriptor: BaseDescriptor) -> None:
+        """
+        Appends descriptor to internal storage.
+        Args:
+            descriptor: descriptor with correct length, version and data
+        Raises:
+            LunaSDKException: if an error occurs while adding the descriptor
+        """
+        error = self._coreIndex.appendDescriptor(descriptor.coreEstimation)
```
Comment (matemax) [inline lunavl/sdk/indexes/builder.py:77]: i think that need check version of descriptor

File: lunavl/sdk/indexes/builder.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        if error.isError:
+            raise LunaSDKException(LunaVLError.fromSDKError(error))
+        self._bufSize += 1
+
+    def appendBatch(self, descriptorsBatch: BaseDescriptorBatch) -> None:
```
Comment (matemax) [inline lunavl/sdk/indexes/builder.py]: i think that need check version of descriptor

File: lunavl/sdk/indexes/builder.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        Args:
+            index: identification of descriptors position in internal storage
+        Raises:
+            IndexError: if index out of range
+            LunaSDKException: if an error occurs while remove descriptor failed
+        """
+        super().__delitem__(index=index)
+        self._bufSize -= 1
+
+    def loadIndex(self, path: str, indexType: IndexType) -> Union[DynamicIndex, DenseIndex]:
```
Comment (matemax) [inline lunavl/sdk/indexes/builder.py:108]: i think that need add descriptor version

File: lunavl/sdk/indexes/stored_index.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+    def append(self, descriptor: BaseDescriptor) -> None:
+        """
+        Appends descriptor to internal storage.
+        Args:
+            descriptor: descriptor with correct length, version and data
+        Raises:
+            LunaSDKException: if an error occurs while adding the descriptor
+        """
+        error = self._coreIndex.appendDescriptor(descriptor.coreEstimation)
```
Comment (matemax) [inline lunavl/sdk/indexes/stored_index.py:41]: i think that need check a version of descriptor

File: tests/test_indexes.py [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+        assert 0.0 < topResult.similarity <= 1.0, "similarity out of range [0,1]"
+
+    def test_search_result_non_default_descriptor(self):
+        """Test search result by descriptor different version."""
+        self.indexBuilder.appendBatch(self.faceDescriptorBatch)
+        dynamicIndex = self.indexBuilder.buildIndex()
+        self.assertDynamicIndex(dynamicIndex, 2, 2)
+        with pytest.raises(LunaSDKException) as ex:
+            dynamicIndex.search(self.nonDefaultFaceDescriptor)
+        self.assertLunaVlError(ex, LunaVLError.InvalidInput.format("Invalid input"))
```
Comment (matemax) [inline tests/test_indexes.py]: need task

File: tests/test_indexes.py [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+        dynamicIndex.append(self.faceDescriptor)
+        self.assertDynamicIndex(dynamicIndex, 1, 1)
+
+    def test_append_descriptor_to_dynamic_index(self):
+        """Test append descriptor to dynamic index."""
+        self.indexBuilder.append(self.faceDescriptor)
+        dynamicIndex = self.indexBuilder.buildIndex()
+        self.assertDynamicIndex(dynamicIndex, 1, 1)
+        dynamicIndex.append(self.faceDescriptor)
+        self.assertDynamicIndex(dynamicIndex, 2, 2)
```
Comment (matemax) [inline tests/test_indexes.py]: need test wth append non default descriptor

File: lunavl/sdk/indexes/stored_index.py [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+
+    def appendBatch(self, descriptorsBatch: BaseDescriptorBatch) -> None:
+        """
+        Appends batch of descriptors to internal storage.
+        Args:
+            descriptorsBatch: Batch of descriptors with correct length, version and data
+        Raises:
+            LunaSDKException: if an error occurs while adding the batch of descriptors
+        """
+        error = self._coreIndex.appendBatch(descriptorsBatch.coreEstimation)
```
Comment (matemax) [inline lunavl/sdk/indexes/stored_index.py:54]: i think that need check a version of descriptor

**Review by matemax — APPROVED** [review]

File: lunavl/sdk/faceengine/engine.py [hunk 8]
```diff
@@ -304,3 +305,23 @@ def createOrientationModeEstimator(self) -> OrientationModeEstimator:
             estimator
         """
         return OrientationModeEstimator(self._faceEngine.createOrientationEstimator())
+
+    def createFaceIndex(self, descriptorVersion: int = 0) -> IndexBuilder:
```
Comment (matemax) [inline lunavl/sdk/faceengine/engine.py]: This function does not create index.

File: examples/sdk_examples/build_index.py [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+    detector = faceEngine.createFaceDetector(DetectorType.FACE_DET_V1)
+    warper = faceEngine.createFaceWarper()
+    extractor = faceEngine.createFaceDescriptorEstimator()
+    descriptorsBatch = faceEngine.createFaceDescriptorFactory().generateDescriptorsBatch(2)
+
+    for image in (EXAMPLE_O, EXAMPLE_1):
+        vlImage = VLImage.load(filename=image)
+        faceDetection = detector.detectOne(vlImage)
+        warp = warper.warp(faceDetection)
+        lastDescriptor = extractor.estimate(warp.warpedImage)
```
Comment (gluhar2006) [inline examples/sdk_examples/build_index.py]: why "last"?

**Review by matemax — APPROVED** [review]

File: lunavl/sdk/indexes/base.py [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+        Args:
+            index: identification of descriptors position in internal storage
+        Raises:
+            IndexError: if index out of range
+            LunaSDKException: if an error occurs while getting descriptor
+        Returns:
+            descriptor
+        """
+        if index >= self.bufSize:
+            raise IndexError(f"Descriptor index '{index}' out of range")    # todo remove after fix FSDK index error
```
Comment (gluhar2006) [inline lunavl/sdk/indexes/base.py]: FSDK task number?

File: lunavl/sdk/indexes/base.py [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+        """
+        Descriptor will be removed from the graph (not from the internal storage), so it is not available for search.
+        Args:
+            index: identification of descriptors position in internal storage
+        Raises:
+            IndexError: if index out of range
+            LunaSDKException: if an error occurs while remove descriptor failed
+        """
+        if index >= self.bufSize:
+            raise IndexError(f"Descriptor index '{index}' out of range")    # todo remove after fix FSDK index error
```
Comment (gluhar2006) [inline lunavl/sdk/indexes/base.py]: task num

File: lunavl/sdk/indexes/builder.py [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+            LunaSDKException: if an error occurs while adding the descriptor
+        """
+        error = self._coreIndex.appendDescriptor(descriptor.coreEstimation)
+        if error.isError:
+            raise LunaSDKException(LunaVLError.fromSDKError(error))
+        self._bufSize += 1
+
+    def appendBatch(self, descriptorsBatch: BaseDescriptorBatch) -> None:
+        """
+        Appends batch of descriptors to internal storage.
```
Comment (gluhar2006) [inline lunavl/sdk/indexes/builder.py:84]: may be append&appendBatch -> append with check (isinstance(input, ...))

File: lunavl/sdk/indexes/builder.py [hunk 13]
```diff
@@ -0,0 +1,116 @@
...
+        super().__delitem__(index=index)
+        self._bufSize -= 1
+
+    def loadIndex(self, path: str, indexType: IndexType) -> Union[DynamicIndex, DenseIndex]:
+        """
+        Load 'dynamic' or 'dense' index from file.
+        Args:
+            path: path to saved index
+            indexType: index type ('dynamic' or 'dense')
+        Raises:
```
Comment (gluhar2006) [inline lunavl/sdk/indexes/builder.py:114]: missing `returns`

File: lunavl/sdk/indexes/builder.py [hunk 14]
```diff
@@ -0,0 +1,116 @@
...
+            _cls = DenseIndex
+        else:
+            raise ValueError(f"{indexType} is not a valid, must be one of ['dynamic', 'dense']")
+        if error.isError:
+            raise LunaSDKException(LunaVLError.fromSDKError(error))
+        return _cls(loadedIndex, self._descriptorFactory)
+
+    def buildIndex(self) -> DynamicIndex:
+        """
+        Builds index with every descriptor appended.
```
Comment (gluhar2006) [inline lunavl/sdk/indexes/builder.py]: with all existing descriptors?

File: lunavl/sdk/indexes/stored_index.py [hunk 15]
```diff
@@ -0,0 +1,115 @@
...
+    def save(self, path: str, indexType: IndexType):
+        """
+        Save index as 'dynamic' or 'dense' to local storage.
+        Args:
+            path: path to file to be created
+            indexType: index type ('dynamic' or 'dense')
+        Raises:
+            LunaSDKException: if an error occurs while saving the index
+        """
+        if Path(path).is_dir():
```
Comment (gluhar2006) [inline lunavl/sdk/indexes/stored_index.py:86]: check directory permissions for writing?

File: lunavl/sdk/indexes/stored_index.py [hunk 16]
```diff
@@ -0,0 +1,115 @@
...
+class DenseIndex(CoreIndex):
+    """
+    Dense Index
+    """
+
+    def __delitem__(self, index: int):
+        """Remove descriptor for a dense index is not supported."""
+        raise AttributeError("'DenseIndex' object has no attribute '__delitem__'")
+
+    def search(self, descriptor: BaseDescriptor, maxCount: Optional[int] = 1) -> List[IndexResult]:
```
Comment (gluhar2006) [inline lunavl/sdk/indexes/stored_index.py]: Optional[int] = 1 ? what's this?

File: tests/test_indexes.py [hunk 17]
```diff
@@ -0,0 +1,290 @@
...
+    defaultFaceEstimator: FaceDescriptorEstimator
+    faceDescriptor: FaceDescriptor
+    nonDefaultFaceDescriptor: FaceDescriptor
+    faceDescriptorBatch: FaceDescriptorBatch
+    indexBuilder: IndexBuilder
+
+    @classmethod
+    def setup_class(cls):
+        super().setup_class()
+        cls.defaultFaceEstimator = BaseTestClass.faceEngine.createFaceDescriptorEstimator(54)
```
Comment (gluhar2006) [inline tests/test_indexes.py]: use EXISTENT_FACE_DESCRIPTOR_VERSION_ABUNDANCE instead of magic numbers

File: tests/test_indexes.py [hunk 18]
```diff
@@ -0,0 +1,290 @@
...
+    def assertDynamicIndex(self, dynamicIndex: DynamicIndex, expectedDescriptorCount: int, expectedBufSize: int):
+        """
+        Assert instance class for dynamic index and check storage size.
+        Args:
+            dynamicIndex: input index
+            expectedDescriptorCount: expected count of descriptors
+            expectedBufSize: expected index buffer size
+        """
+        assert isinstance(dynamicIndex, DynamicIndex), f"created {dynamicIndex} is not {DynamicIndex}"
+        assert isinstance(dynamicIndex.bufSize, int), f"expected int but found {dynamicIndex.bufSize}"
```
Comment (gluhar2006) [inline tests/test_indexes.py]: ... but found value? use `__class__`

File: tests/test_indexes.py [hunk 19]
```diff
@@ -0,0 +1,290 @@
...
+    def getCountOfDescriptorsInStorage(indexObject: Union[IndexBuilder, DynamicIndex, DenseIndex]) -> int:
+        """
+        Get actual count of descriptor in internal storage.
+        Args:
+            indexObject: an initialized index object with `getDescriptor` method
+        Returns:
+            count of descriptors
+        """
+        count = 0
+        while True:
```
Comment (gluhar2006) [inline tests/test_indexes.py:94]: very very strange. what's this?

Comment (gluhar2006) [reply tests/test_indexes.py:94]: seems it needs to remove after fsdk fix bug?

File: tests/test_indexes.py [hunk 20]
```diff
@@ -0,0 +1,290 @@
...
+        for idx in range(expectedDescriptorsCount):
+            with self.subTest(case=f"get descriptor with index: {idx}"):
+                descriptor = self.indexBuilder[idx]
+                assert self.faceDescriptorBatch[idx].rawDescriptor == descriptor.rawDescriptor
+
+    def test_get_descriptor_from_builder_bad_index(self):
+        """Test get descriptor with invalid index from internal storage."""
+        self.indexBuilder.appendBatch(self.faceDescriptorBatch)
+        assert len(self.faceDescriptorBatch) == self.indexBuilder.bufSize
+        nonexistentIndex = 2
```
Comment (gluhar2006) [inline tests/test_indexes.py]: camelCase

File: tests/test_indexes.py [hunk 21]
```diff
@@ -0,0 +1,290 @@
...
+        with self.assertRaises(IndexError) as ex:
+            descriptor = self.indexBuilder[nonexistentIndex]
+        assert str(nonexistentIndex) in ex.exception.args[0]
```
Comment (gluhar2006) [inline tests/test_indexes.py]: will this test be relevant after the fsdk fix?

Comment (gluhar2006) [reply tests/test_indexes.py]: can we assert Exception?

File: tests/test_indexes.py [hunk 22]
```diff
@@ -0,0 +1,290 @@
...
+
+    def test_get_non_default_descriptor_from_builder(self):
+        """Test get non default descriptor from internal storage."""
+        indexBuilder = self.faceEngine.createFaceIndex(descriptorVersion=56)
+        indexBuilder.append(self.faceDescriptor)
+        assert 1 == indexBuilder.bufSize
+        with pytest.raises(LunaSDKException) as ex:
+            descriptor = indexBuilder[0]
+        self.assertLunaVlError(ex, LunaVLError.InvalidInput.format("Invalid input"))
```
Comment (gluhar2006) [inline tests/test_indexes.py]: will this assert be relevant after the fsdk fix?

File: tests/test_indexes.py [hunk 23]
```diff
@@ -0,0 +1,290 @@
...
+
+    def test_remove_descriptor_from_builder(self):
+        """Test remove descriptor from internal storage."""
+        expectedDescriptorsCount = 2
+        self.indexBuilder.appendBatch(self.faceDescriptorBatch)
+        assert expectedDescriptorsCount == self.indexBuilder.bufSize
+        for index, expectedBufSize in ((1, 1), (0, 0)):
```
Comment (gluhar2006) [inline tests/test_indexes.py:161]: index equals bufSize? why do we need tuple?

File: tests/test_indexes.py [hunk 24]
```diff
@@ -0,0 +1,290 @@
...
+        dynamicIndex = self.indexBuilder.buildIndex()
+        self.assertDynamicIndex(dynamicIndex, 1, 1)
+        dynamicIndex.append(self.faceDescriptor)
+        self.assertDynamicIndex(dynamicIndex, 2, 2)
+
+    def test_append_descriptors_batch_to_dynamic_index(self):
+        """Test append descriptor batch to dynamic index."""
+        self.indexBuilder.append(self.faceDescriptor)
+        dynamicIndex = self.indexBuilder.buildIndex()
+        self.assertDynamicIndex(dynamicIndex, 1, 1)
```
Comment (gluhar2006) [inline tests/test_indexes.py]: use keyword please

File: lunavl/sdk/indexes/builder.py (hunk 1 above)
Comment (DeusAnimo) [reply lunavl/sdk/indexes/builder.py:77]: added and opened fsdk issue

File: lunavl/sdk/indexes/builder.py (hunk 2 above)
Comment (DeusAnimo) [reply lunavl/sdk/indexes/builder.py]: added and opened fsdk issue

File: lunavl/sdk/indexes/builder.py (hunk 3 above)
Comment (DeusAnimo) [reply lunavl/sdk/indexes/builder.py:108]: added and opened fsdk issue

File: lunavl/sdk/indexes/stored_index.py (hunk 4 above)
Comment (DeusAnimo) [reply lunavl/sdk/indexes/stored_index.py:41]: added and opened fsdk issue

File: tests/test_indexes.py (hunk 5 above)
Comment (DeusAnimo) [reply tests/test_indexes.py]: added

File: tests/test_indexes.py (hunk 6 above)
Comment (DeusAnimo) [reply tests/test_indexes.py]: updated

File: lunavl/sdk/indexes/stored_index.py (hunk 7 above)
Comment (DeusAnimo) [reply lunavl/sdk/indexes/stored_index.py:54]: added and opened fsdk issue

File: lunavl/sdk/faceengine/engine.py (hunk 8 above)
Comment (DeusAnimo) [reply lunavl/sdk/faceengine/engine.py]: renamed to index builder

File: examples/sdk_examples/build_index.py (hunk 9 above)
Comment (DeusAnimo) [reply examples/sdk_examples/build_index.py]: fixed

File: lunavl/sdk/indexes/base.py (hunk 10 above)
Comment (DeusAnimo) [reply lunavl/sdk/indexes/base.py]: fixed, was opened issue

File: lunavl/sdk/indexes/base.py (hunk 11 above)
Comment (DeusAnimo) [reply lunavl/sdk/indexes/base.py]: fixed

File: lunavl/sdk/indexes/builder.py (hunk 12 above)
Comment (DeusAnimo) [reply lunavl/sdk/indexes/builder.py:84]: agree, updated

File: lunavl/sdk/indexes/builder.py (hunk 13 above)
Comment (DeusAnimo) [reply lunavl/sdk/indexes/builder.py:114]: added

File: lunavl/sdk/indexes/stored_index.py (hunk 15 above)
Comment (DeusAnimo) [reply lunavl/sdk/indexes/stored_index.py:86]: checking write access was added

File: lunavl/sdk/indexes/builder.py (hunk 14 above)
Comment (DeusAnimo) [reply lunavl/sdk/indexes/builder.py]: yep, fixed

File: lunavl/sdk/indexes/stored_index.py (hunk 16 above)
Comment (DeusAnimo) [reply lunavl/sdk/indexes/stored_index.py]: fixed, not optional. that's associated with top result

File: tests/test_indexes.py (hunk 17 above)
Comment (DeusAnimo) [reply tests/test_indexes.py]: fixed

File: tests/test_indexes.py (hunk 18 above)
Comment (DeusAnimo) [reply tests/test_indexes.py]: my mistake (

File: tests/test_indexes.py (hunk 23 above)
Comment (DeusAnimo) [reply tests/test_indexes.py:161]: fixed

File: tests/test_indexes.py (hunk 24 above)
Comment (DeusAnimo) [reply tests/test_indexes.py]: ok, updated

File: tests/test_indexes.py (hunk 20 above)
Comment (DeusAnimo) [reply tests/test_indexes.py]: fixed

File: tests/test_indexes.py (hunk 21 above)
Comment (DeusAnimo) [reply tests/test_indexes.py]: maybe we should leave this exception and not rely on fsdk? I have no idea how they will fix it :(

File: lunavl/sdk/indexes/builder.py [hunk 25]
```diff
@@ -0,0 +1,109 @@
...
+        elif isinstance(descriptor, FaceDescriptorBatch):
+            appendDescriptor = self._coreIndex.appendBatch
+        else:
+            raise RuntimeError(f"Not supported descriptor class: {descriptor.__class__}")
+
+        self.checkDescriptorVersion(descriptor)
+        error = appendDescriptor(descriptor.coreEstimation)
+        if error.isError:
+            raise LunaSDKException(LunaVLError.fromSDKError(error))
+        self._bufSize += len(descriptor) if hasattr(descriptor, "__len__") else 1
```
Comment (gluhar2006) [inline lunavl/sdk/indexes/builder.py]: may be, len() if isinstance(descriptor<s?>), list)?

File: lunavl/sdk/indexes/stored_index.py [hunk 26]
```diff
@@ -0,0 +1,119 @@
...
+        return self._coreIndex.countOfIndexedDescriptors()
+
+    def append(self, descriptor: Union[FaceDescriptor, FaceDescriptorBatch]) -> None:
+        """
+        Appends descriptor to internal storage.
+        Args:
+            descriptor: descriptor or batch of descriptors with correct length, version and data
+        Raises:
+            RuntimeError: if descriptor type is not supported
+            LunaSDKException: if an error occurs while adding the descriptor
```
Comment (gluhar2006) [inline lunavl/sdk/indexes/stored_index.py:38]: outdated?

File: lunavl/sdk/indexes/stored_index.py [hunk 27]
```diff
@@ -0,0 +1,119 @@
...
+        if Path(path).is_dir():
+            raise ValueError(f"{path} must not be a directory")
+        if not os.access(Path(path).parent, os.W_OK):
+            raise PermissionError(f"Access is denied: {path}")
+        if indexType == IndexType.dynamic:
+            error = self._coreIndex.saveToDynamicIndex(path)
+        elif indexType == IndexType.dense:
+            error = self._coreIndex.saveToDenseIndex(path)
+        else:
+            raise ValueError(f"{indexType} is not a valid, must be one of ['dynamic', 'dense']")
```
Comment (gluhar2006) [inline lunavl/sdk/indexes/stored_index.py]: use IndexType enum?

File: lunavl/sdk/indexes/builder.py [hunk 28]
```diff
@@ -0,0 +1,110 @@
...
+        Args:
+            descriptor: descriptor or batch of descriptors with correct length, version and data
+        Raises:
+            RuntimeError: if descriptor type is not supported
+            LunaSDKException: if an error occurs while adding the descriptor
+        """
+        if isinstance(descriptor, FaceDescriptor):
+            appendDescriptor = self._coreIndex.appendDescriptor
+        elif isinstance(descriptor, FaceDescriptorBatch):
+            appendDescriptor = self._coreIndex.appendBatch
```
Comment (matemax) [inline lunavl/sdk/indexes/builder.py]: extend?

File: lunavl/sdk/indexes/stored_index.py [hunk 29]
```diff
@@ -0,0 +1,119 @@
...
+        Args:
+            descriptor: descriptor or batch of descriptors with correct length, version and data
+        Raises:
+            RuntimeError: if descriptor type is not supported
+            LunaSDKException: if an error occurs while adding the descriptor
+        """
+        if isinstance(descriptor, FaceDescriptor):
+            appendDescriptor = self._coreIndex.appendDescriptor
+        elif isinstance(descriptor, FaceDescriptorBatch):
+            appendDescriptor = self._coreIndex.appendBatch
```
Comment (matemax) [inline lunavl/sdk/indexes/stored_index.py]: extend?very strange that descriptor has type FaceDescriptorBatch

File: lunavl/sdk/indexes/base.py [hunk 30]
```diff
@@ -0,0 +1,145 @@
...
+        Check descriptor version with version in index.
+        todo: remove this function after normalizing errors (FSDK-2897)
+        Args:
+            receivedDescriptorVersion: received descriptor version
+        Raises:
+            LunaSDKException(InvalidDescriptor): if descriptor versions do not match
+        """
+        if receivedDescriptorVersion != self.descriptorVersion:
+            raise LunaSDKException(
+                LunaVLError.InvalidDescriptor.format(f"Not expected descriptor version {receivedDescriptorVersion}")
```
Comment (gluhar2006) [inline lunavl/sdk/indexes/base.py]: may be smth like `Expected descriptor version 1, got version 2`?

File: lunavl/sdk/indexes/builder.py [hunk 31]
```diff
@@ -0,0 +1,112 @@
...
+    Attributes:
+        _bufSize (int): storage size with descriptors
+    """
+
+    def __init__(self, faceEngine: PyIFaceEngine):
+        super().__init__(faceEngine.createIndexBuilder(), faceEngine)
+        self._bufSize = 0
+
+    @property
+    def bufSize(self) -> int:
```
Comment (gluhar2006) [inline lunavl/sdk/indexes/builder.py:32]: for what?

Comment (DeusAnimo) [reply lunavl/sdk/indexes/builder.py:32]: to get information about the size of index (other indexes have this information)

File: lunavl/sdk/indexes/base.py (hunk 30 above)
Comment (DeusAnimo) [reply lunavl/sdk/indexes/base.py]: fixed

**Review by gluhar2006 — APPROVED** [review]

File: lunavl/sdk/indexes/builder.py (hunk 28 above)
Comment (DeusAnimo) [reply lunavl/sdk/indexes/builder.py]: fixed

File: lunavl/sdk/indexes/stored_index.py (hunk 29 above)
Comment (DeusAnimo) [reply lunavl/sdk/indexes/stored_index.py]: fixed

**Pull request merged**